datahub-web-react/src/app/search/EntitySearchResults.tsx

```diff
@@ -0,0 +1,133 @@
+import { SearchFilters } from './SearchFilters';
+
+const styles = {
+    loading: { marginTop: '10%' },
+    addFilters: { backgroundColor: '#F5F5F5' },
+    resultSummary: { color: 'gray', marginTop: '36px' },
+    resultHeaderCardBody: { padding: '16px 24px' },
+    resultHeaderCard: { right: '52px', top: '-40px', position: 'absolute' },
+    resultList: { width: '100%', borderColor: '#f0f0f0', marginTop: '12px', padding: '16px 32px' },
+    paginationRow: { padding: 40 },
+    resultsContainer: { width: '100%', padding: '20px 132px' },
+};
+
+interface Props {
+    type: EntityType;
+    query: string;
+    page: number;
+    filters: Array<FacetFilterInput>;
+    onChangeFilters: (filters: Array<FacetFilterInput>) => void;
+    onChangePage: (page: number) => void;
+}
+
+export const EntitySearchResults = ({ type, query, page, filters, onChangeFilters, onChangePage }: Props) => {
+    const [isEditingFilters, setIsEditingFilters] = useState(false);
+    const [selectedFilters, setSelectedFilters] = useState(filters);
+    useEffect(() => {
+        setSelectedFilters(filters);
+    }, [filters]);
+
+    const entityRegistry = useEntityRegistry();
+    const { loading, error, data } = useGetSearchResultsQuery({
+        variables: {
+            input: {
+                type,
+                query,
+                start: (page - 1) * SearchCfg.RESULTS_PER_PAGE,
+                count: SearchCfg.RESULTS_PER_PAGE,
+                filters,
+            },
+        },
+    });
+
+    const results = data?.search?.entities || [];
+    const pageStart = data?.search?.start || 0;
+    const pageSize = data?.search?.count || 0;
+    const totalResults = data?.search?.total || 0;
+    const lastResultIndex =
+        pageStart * pageSize + pageSize > totalResults ? totalResults : pageStart * pageSize + pageSize;
+
+    const onFilterSelect = (selected: boolean, field: string, value: string) => {
+        const newFilters = selected
+            ? [...selectedFilters, { field, value }]
+            : selectedFilters.filter((filter) => filter.field !== field || filter.value !== value);
+        setSelectedFilters(newFilters);
+    };
+
+    const onEditFilters = () => {
+        setIsEditingFilters(true);
+    };
+
+    const onCloseEditFilters = () => {
+        onChangeFilters(selectedFilters);
+        setIsEditingFilters(false);
+    };
+
+    if (error || (!loading && !error && !data)) {
+        return <Alert type="error" message={error?.message || 'Entity failed to load'} />;
+    }
+
+    return (
+        <div style={styles.resultsContainer}>
+            {loading && <Message type="loading" content="Loading..." style={styles.loading} />}
+            <Button style={styles.addFilters} onClick={onEditFilters}>
+                <FilterOutlined />
+                Add Filters
+            </Button>
+            <Modal
+                title="Filters"
+                footer={<Button onClick={onCloseEditFilters}>Apply</Button>}
```

Comment:
should we support a cancel too so they have an option to dismiss their staged changes?

Reply:
sure we can support the "x" button